views/topologicalSort.pug

```diff
@@ -1,9 +1,263 @@
+                        table.append("tr")
+                            .attr("bgcolor", bgcolor)
+                            .selectAll("td")
+                            .data(td_data)
+                            .enter()
+                            .append("td")
+                            .text(function(d,i){return d})
+                        if(path[i] == mid_name){
+                            bgcolor = "green"
+                        }
+                        break
+                    }
+                }
+            }
+        }
+        // draw svg & axis
+        var width = 550; 
+        var height = 550; 
+
+        var svg = d3.select("body")            
+                    .append("svg")              
+                    .attr("width", width)       
+                    .attr("height", height); 
+        var dataset = [ 0,1,2,3,4,5,6,7,8,9,10];
+        var linearX = d3.scale.linear()
+                .domain([0, d3.max(dataset)])
+                .range([0, 500]);
+        var linearY = d3.scale.linear()
+                .domain([0, d3.max(dataset)])
+                .range([500, 0]);
+        var axisX = d3.svg.axis()
+                    .scale(linearX)      //指定比例尺
+                    .orient("bottom")   //指定刻度的方向
+                    .ticks(11);          //指定刻度的数量
+        svg.append("g")
+            .attr("class","axis")
+            .attr("transform","translate(20,520)")
+            .call(axisX);
+        var axisY = d3.svg.axis()
+                    .scale(linearY) 
+                    .orient("left")
+                    .ticks(11);     
+        svg.append("g")
+            .attr("class","axis")
+            .attr("transform","translate(20,20)")
+            .call(axisY);
+
+        // get data
+        var vertex_dict = {}
+        var points_set = new Array()
+        var data = !{JSON.stringify(data)}
+        var csvdata = data["csvdata"]
+        var threLower = data["threLower"]
+        var threUpper = data["threUpper"]
+        console.log(threLower)
+
+        
+
+        for(var i=0;i<csvdata.length;i++){
+            var t_name = csvdata[i]['id']
+            var t_point = [csvdata[i]['RRR\''] * 1,csvdata[i]['T\'c'] * 1]
+            vertex_dict[t_name] = t_point
+            var point_dic = {"point":t_point, "name":t_name}
+            points_set.push(point_dic)
+        }
+    
+        // draw points
+        svg.selectAll("circle")
+            .data(points_set)
+            .enter()
+            .append("circle")
+            .attr("stroke","blue")
+            .attr("stroke-width","1")
+            .attr("r","5")
+            .attr("fill","blue")
+            .attr("id",function(d,i){return d["name"]})
+            .attr("cx",function(d,i){
+                return d["point"][0]*50 + 20
+            })
+            .attr("cy",function(d,i){
+                return 520 - d["point"][1]*50
+            })
+
+        // set mid_point
+        mid_name = "Filet-O-Fish"
```

Comment:
Can this work when using hpb data? Or you mean mid_name = '' ? Sorry for missing it yesterday.

Reply:
It doesn't matter. But thanks, I should delete this line to prevent misunderstanding.